### DISCUSSION TITLE
CHANGE PINS OF MOSI MISO SCK SS

### DISCUSSION THREAD

**Discussion by RAFIz11** [khoih-prog/RP2040_SD#6] [General]
HELLO
I NEED TO CHANGE THIS PINS TO
GP4 MISO
GP5 CSN
GP6 SCK
GP7 MOSI
ITS THE DEFAULT OUTPUT BY RASPBERRY SO I NEED THOES OUTPUT AND NOT THAT DEFAULT
CAN YOU TELL ME WHERE I CAN EDIT THAT
PLS HELP ME ITS FOR MY UNIVESIRY PROJECT
Finally i fix the problem

**Comment by janihorvath** [comment]
Hello,
I have a similar question. I've just received a Cytron Maker Pi Pico board with on-board SD card module (https://www.cytron.io/p-maker-pi-pico). The SD card module works with CircuitPython but I'm much more familiar with the Arduino IDE and also want to move my IMU project from a Nano to the Pico with minor adjustments.
The wiring is a little bit different so I tried the following: 

```
#if !defined(ARDUINO_ARCH_RP2040)
  #error For RP2040 only
#endif

#if defined(ARDUINO_ARCH_MBED)
  
  #define PIN_SD_MOSI       PIN_SPI_MOSI
  #define PIN_SD_MISO       PIN_SPI_MISO
  #define PIN_SD_SCK        PIN_SPI_SCK
  #define PIN_SD_SS         PIN_SPI_SS

#else

**//I manually changed here the pins:**

  #define PIN_SD_MOSI       11 
  #define PIN_SD_MISO       12
  #define PIN_SD_SCK        10
  #define PIN_SD_SS         15
  
#endif

#define _RP2040_SD_LOGLEVEL_       4
```

After uploading the following error message is printed to the serial: "Initialization failed"

How could I make this work?
Thank you in advance!
